File: python/mail.py
# -*- coding: UTF-8 -*-

import yagmail


def sendMail():
    # 设置发送邮件的目标邮箱
    email = '修改为目标邮箱'

    # 通过自己的邮箱向用户邮箱发送邮件
    yag = yagmail.SMTP(user='修改为发送邮箱', password='发送邮箱的密码或权限码', host='smtp.tongji.edu.cn')

    # 发送邮件
    subject = "程序完成通知"
    message = "您的程序已运行完成"

    yag.send(to=[email], subject=subject, contents=message)
    print("邮件通知已发送")

File: matlab/mail.m
function mail()
    mail = '发送邮箱';
    password = '对应邮箱的密码和权限码';
    
    setpref('Internet', 'SMTP_Server', 'smtp.qq.com');
    setpref('Internet', 'E_mail', mail);
    setpref('Internet', 'SMTP_Username', mail);
    setpref('Internet', 'SMTP_Password', password);

    props = java.lang.System.getProperties;
    props.setProperty('mail.smtp.auth','true');
    props.setProperty('mail.smtp.socketFactory.class', 'javax.net.ssl.SSLSocketFactory');
    props.setProperty('mail.smtp.socketFactory.port','465');
    
    recipient = '目标邮箱';
    subject = '任务完成提醒';
    message = '您的代码已运行结束';
    
    sendmail(recipient, subject, message);
end